src/CreateSignalFactory.js

```diff
@@ -1,15 +1,18 @@
 var utils = require('./utils.js')
-var createActionArgs = require('./createActionArgs.js')
-var createNext = require('./createNext.js')
+// var createNext = require('./createNext.js')
 var analyze = require('./analyze.js')
 var staticTree = require('./staticTree')
-var createModulesArg = require('./createModulesArg.js')
+var createContext = require('./createContext')
+var inputProvider = require('./contextProviders/inputProvider')
+var stateProvider = require('./contextProviders/stateProvider')
+var servicesProvider = require('./contextProviders/servicesProvider')
+var outputProvider = require('./contextProviders/outputProvider')
 
 var requestAnimationFrame = global.requestAnimationFrame || function (cb) {
   setTimeout(cb, 0)
 }
 
-module.exports = function (controller, model, services, compute, modules) {
+module.exports = function (controller, model, services, compute, modules, externalContextProviders) {
```

Comment:
`model` could be now got from `contoller.getModel()`
`services, compute, modules` is not used anymore :)